File: api/src/lib/application/http/user/validators.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;
use uuid::Uuid;
use validator::Validate;

#[derive(Debug, Serialize, Deserialize, Validate, ToSchema)]
pub struct ResetPasswordValidator {
    #[serde(default)]
    pub temporary: bool,

    #[serde(default)]
    pub credential_type: String,

    #[validate(length(min = 1, message = "value is required"))]
    #[serde(default)]
    pub value: String,
}

#[derive(Debug, Serialize, Deserialize, Validate, ToSchema)]
pub struct CreateUserValidator {
    #[validate(length(min = 1, message = "username is required"))]
    #[serde(default)]
    pub username: String,

    #[validate(length(min = 1, message = "firstname is required"))]
    #[serde(default)]
    pub firstname: String,

    #[validate(length(min = 1, message = "lastname is required"))]
    #[serde(default)]
    pub lastname: String,

    #[validate(length(min = 1, message = "email is required"))]
    #[serde(default)]
    pub email: String,

    #[serde(default)]
    pub email_verified: Option<bool>,
}

#[derive(Debug, Serialize, Deserialize, Validate, ToSchema)]
pub struct BulkDeleteUserValidator {
    #[serde(default)]
    pub ids: Vec<Uuid>,
}

File: front/src/pages/user/validators.ts
import { z } from "zod"

export const createUserValidator = z.object({
  username: z.string().min(1),
  firstname: z.string().min(1),
  lastname: z.string().min(1),
  email: z.string().email().min(1),
  email_verified: z.boolean().optional(),
})

export const updateUserValidator = z.object({
  username: z.string().min(1),
  firstname: z.string().min(1),
  lastname: z.string().min(1),
  enabled: z.boolean().optional(),
  email: z.string().email().min(1),
  email_verified: z.boolean().optional(),
  required_actions: z.array(z.string()).optional(),
})


export type CreateUserSchema = z.infer<typeof createUserValidator>
export type UpdateUserSchema = z.infer<typeof updateUserValidator>